Compile Error using non standard compile options

for the current dev and the LaserWakefield example I am getting a compile error, when using the following pic-build options
`pic-build -b serial -c "-DPIC_CI_COMPILE=ON -DCMAKE_BUILD_TYPE=Debug" 2>&1 | tee compile.result`

```
[ 97%] Building CXX object CMakeFiles/picongpu.dir/versionFormat.cpp.o
In file included from /home/marre55/picongpu/include/picongpu/../pmacc/kernel/operation/Atomic.hpp:24,
                 from /home/marre55/picongpu/include/picongpu/../picongpu/fields/currentDeposition/Strategy.def:24,
                 from /home/marre55/picongpu/include/picongpu/../picongpu/fields/currentDeposition/EmZ/EmZ.def:24,
                 from /home/marre55/picongpu/include/picongpu/../picongpu/fields/currentDeposition/Solver.def:21,
                 from /home/marre55/picongpu/include/picongpu/../picongpu/param/species.param:34,
                 from /home/marre55/picongpu/include/picongpu/../picongpu/_defaultParam.loader:53,
                 from /home/marre55/picongpu/include/picongpu/../picongpu/simulation_defines.hpp:44,
                 from /home/marre55/picongpu/include/picongpu/main.cpp:23:
In member function 'T_Type pmacc::kernel::detail::AtomicAllInc<T_Type, T_isKepler>::operator()(const T_Acc&, T_Type*, const T_Hierarchy&) [with T_Acc = alpaka::AccCpuSerial<std::integral_constant<long unsigned int, 3>, unsigned int>; T_Hierarchy = alpaka::hierarchy::Grids; T_Type = long long unsigned int; bool T_isKepler = false]',
    inlined from 'T pmacc::kernel::atomicAllInc(T*) [with T = long long unsigned int]' at /home/marre55/picongpu/include/picongpu/../pmacc/kernel/atomic.hpp:167:89,
    inlined from 'static uint64_t pmacc::IdProvider<T_dim>::getNewId() [with unsigned int T_dim = 3]' at /home/marre55/picongpu/include/picongpu/../pmacc/particles/IdProvider.hpp:116:58,
    inlined from 'uint64_t pmacc::IdProvider<T_dim>::GetNewId::operator()() const [with unsigned int T_dim = 3]' at /home/marre55/picongpu/include/picongpu/../pmacc/particles/IdProvider.def:66:32,
    inlined from 'void pmacc::idDetail::KernelGetNewId::operator()(const T_Acc&, T_Box, T_IdFactory) const [with T_Box = pmacc::DataBox<pmacc::PitchedBox<long long unsigned int, 1> >; T_IdFactory = pmacc::IdProvider<3>::GetNewId; T_Acc = alpaka::AccCpuSerial<std::integral_constant<long unsigned int, 3>, unsigned int>]' at /home/marre55/picongpu/include/picongpu/../pmacc/particles/IdProvider.hpp:61:38,
    inlined from 'constexpr _Res std::__invoke_impl(__invoke_other, _Fn&&, _Args&& ...) [with _Res = void; _Fn = const pmacc::idDetail::KernelGetNewId&; _Args = {alpaka::AccCpuSerial<integral_constant<long unsigned int, 3>, unsigned int>&, pmacc::DataBox<pmacc::PitchedBox<long long unsigned int, 1> >, pmacc::IdProvider<3>::GetNewId}]' at /opt/spack-modules/linux-ubuntu20.04-skylake_avx512/gcc-10.2.0/gcc-12.2.0-nupdqwabpymog72rxix6bong2zqyms6a/lib/gcc/x86_64-pc-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/invoke.h:61:36:
/home/marre55/picongpu/include/picongpu/../pmacc/kernel/atomic.hpp:47:67: error: 'fakeAcc' may be used uninitialized [-Werror=maybe-uninitialized]
   47 |                     return ::alpaka::atomicOp<::alpaka::AtomicAdd>(acc, ptr, T_Type(1), hierarchy);
      |                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/marre55/picongpu/thirdParty/cupla/alpaka/include/alpaka/atomic/AtomicAtomicRef.hpp:7,
                 from /home/marre55/picongpu/thirdParty/cupla/alpaka/include/alpaka/atomic/AtomicCpu.hpp:13,
                 from /home/marre55/picongpu/thirdParty/cupla/alpaka/include/alpaka/acc/AccCpuOmp2Blocks.hpp:8,
                 from /home/marre55/picongpu/thirdParty/cupla/alpaka/include/alpaka/alpaka.hpp:13,
                 from /home/marre55/picongpu/thirdParty/cupla/include/cupla/defines.hpp:25,
                 from /home/marre55/picongpu/thirdParty/cupla/include/cupla/types.hpp:23,
                 from /home/marre55/picongpu/include/picongpu/../pmacc/types.hpp:30,
                 from /home/marre55/picongpu/include/picongpu/../picongpu/simulation_defines.hpp:26:
/home/marre55/picongpu/thirdParty/cupla/alpaka/include/alpaka/atomic/Traits.hpp: In function 'constexpr _Res std::__invoke_impl(__invoke_other, _Fn&&, _Args&& ...) [with _Res = void; _Fn = const pmacc::idDetail::KernelGetNewId&; _Args = {alpaka::AccCpuSerial<integral_constant<long unsigned int, 3>, unsigned int>&, pmacc::DataBox<pmacc::PitchedBox<long long unsigned int, 1> >, pmacc::IdProvider<3>::GetNewId}]':
/home/marre55/picongpu/thirdParty/cupla/alpaka/include/alpaka/atomic/Traits.hpp:71:29: note: by argument 1 of type 'const alpaka::AccCpuSerial<std::integral_constant<long unsigned int, 3>, unsigned int>&' to 'T alpaka::atomicOp(const TAtomic&, T*, const T&, const THierarchy&) [with TOp = AtomicAdd; TAtomic = AccCpuSerial<std::integral_constant<long unsigned int, 3>, unsigned int>; T = long long unsigned int; THierarchy = hierarchy::Grids]' declared here
   71 |     ALPAKA_FN_HOST_ACC auto atomicOp(
      |                             ^~~~~~~~
/home/marre55/picongpu/include/picongpu/../pmacc/kernel/atomic.hpp:166:58: note: 'fakeAcc' declared here
  166 |             pmacc::memory::Array<cupla::AccThreadSeq, 1> fakeAcc;
      |                                                          ^~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/picongpu.dir/build.make:76: CMakeFiles/picongpu.dir/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:252: CMakeFiles/picongpu.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

ERROR: Could not successfully run make install in build directory:
       .build
```
[compile.result](https://github.com/ComputationalRadiationPhysics/picongpu/files/13938923/compile.result.txt)

the compile error vanishes if I remove the `-DCMAKE_BUILD_TYPE=Debug` compile option

--- a/include/pmacc/kernel/atomic.hpp
+++ b/include/pmacc/kernel/atomic.hpp
@@ -163,9 +163,9 @@ namespace pmacc
              * The id provider for particles is the only code where atomicAllInc is used without an accelerator.
              * @todo remove the unsafe faked accelerator
              */
-            pmacc::memory::Array<cupla::AccThreadSeq, 1> fakeAcc;
+            pmacc::memory::Array<std::byte, sizeof(cupla::AccThreadSeq)> fakeAcc(std::byte(0));
             return detail::AtomicAllInc<T, (PMACC_CUDA_ARCH >= 300 || BOOST_COMP_HIP)>()(
-                fakeAcc[0],
+                *reinterpret_cast<cupla::AccThreadSeq*>(fakeAcc.data()),
                 ptr,
                 ::alpaka::hierarchy::Grids());
         }

